utilities/os_utils.py

```
@@ -0,0 +1,244 @@
+CENTOS_OS_MAPPING: dict[str, dict[str, str | Any]] = {
+    WORKLOAD_STR: Template.Workload.SERVER,
+    FLAVOR_STR: Template.Flavor.TINY,
+    "centos-stream-9": {
+        IMAGE_NAME_STR: "CENTOS_STREAM_9_IMG",
+        OS_VERSION_STR: "9",
+        OS_STR: "centos-stream9",
+    },
+}
+
+
+def generate_os_matrix_dict(os_name: str, supported_operating_systems: list[str]) -> list[dict[str, Any]]:
```

Pylint (3.3.7)

[refactor] Too many branches (14/12)

(R0912)